### ISSUE TITLE
Heuristic search options are not checking is "search=heur"

### ISSUE THREAD

**Issue by cyrille-artho** [cyrille-artho/modbat#108]
It's currently possible to set a lot of search options without them having any effect.
The options should use a `@Requires` annotation so a user does not set them when they are not available.
See `config/src/main/scala/modbat/config/TestConfiguration.scala` for examples and `src/main/scala/modbat/mbt/Configuration.scala` for the actual configuration.

**Comment by wruiwr** [comment]
It seems that it just need to add @Requires(opt = "search", equals = "heur") for each reward variable. 🤔


 @Doc("search mode (for usage mode=exec)")
 @Choice(Array("random", "heur"))
  var search = "random"

  @Doc("bandit trade off value (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var banditTradeoff = 2

  @Doc("backtrack transition reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var backtrackTReward = 0.8d

  @Doc("self-loop transition reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var selfTReward = 0.4d

  @Doc("good and successful transition reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var goodTReward = 0.6d

  @Doc("failed transition reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var failTReward = 0.5d

  @Doc("passed precondition reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var precondPassReward = 0.7d

  @Doc("failed precondition reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var precondFailReward = 0.7d

  @Doc("passed assertion reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var assertPassReward = 0.7d

  @Doc("failed assertion reward (for usage search=heur)")
  @Requires(opt = "search", equals = "heur")
  var assertFailReward = 0.7d

**Comment by cyrille-artho** [comment]
Yes, I agree that this is easy to add. However, we need to check if any existing scripts (from our papers) break if we enforce this. Otherwise, it's a good beginner task to someone who wants to make a first contribution. Your comment is basically already a patch, so perhaps a newcomer can figure out which file to apply it to and how to create a PR. :-)
